operator/src/trustee.rs

```diff
@@ -181,31 +182,73 @@ pub async fn generate_reference_values(
     Ok(())
 }
 
-// TODO: this function needs to be removed, right now it is only for testing the resource
-pub async fn generate_secret(client: Client, namespace: &str, name: &str) -> anyhow::Result<()> {
-    let mut data = BTreeMap::new();
-    data.insert(
-        "key1".to_string(),
-        k8s_openapi::ByteString(b"test".to_vec()),
-    );
+fn random_bytes() -> anyhow::Result<[u8; 32]> {
+    let mut pass = [0; 32];
+    openssl::rand::rand_bytes(&mut pass)?;
+    Ok(pass)
+}
+
+pub async fn generate_secret(
+    client: Client,
+    namespace: &str,
+    kbs_config_name: &str,
+    id: &str,
+) -> anyhow::Result<()> {
+    let pass = random_bytes()?;
+    let secret_data = k8s_openapi::ByteString(pass.to_vec());
+    let data = BTreeMap::from([("root".to_string(), secret_data)]);
 
     let secret = Secret {
         metadata: kube::api::ObjectMeta {
-            name: Some(name.to_string()),
+            name: Some(id.to_string()),
             namespace: Some(namespace.to_string()),
             ..Default::default()
         },
         data: Some(data),
         ..Default::default()
     };
 
-    let secrets: Api<Secret> = Api::namespaced(client, namespace);
+    let secrets: Api<Secret> = Api::namespaced(client.clone(), namespace);
     match secrets.create(&PostParams::default(), &secret).await {
         Ok(s) => info!("Created Secret {:?}", s.metadata.name),
-        Err(Error::Api(ae)) if ae.code == 409 => info!("Secret {} already exists", name),
+        Err(Error::Api(ae)) if ae.code == 409 => info!("Secret {} already exists", id),
         Err(e) => return Err(e.into()),
     }
 
+    let kbs_configs: Api<KbsConfig> = Api::namespaced(client, namespace);
+
+    let existing_secrets = kbs_configs
+        .get(kbs_config_name)
+        .await?
+        .spec
+        .kbs_secret_resources;
+    if existing_secrets.iter().any(|s| s == id) {
+        return Err(anyhow!("Secret with ID {id} already present"));
```

Comment:
Not sure, we really need to error here. It depends on how you react to this error, but I would simply return without error

Reply:
updated both